src/components/Todos.jsx

```diff
@@ -0,0 +1,220 @@
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import Todolist from "./Todolist";
+import RttIcon from "@mui/icons-material/Rtt";
+import { v4 as uuidv4 } from "uuid";
+import { getLocalStorage } from "../services/localStorage";
+
+function Todos() {
+  const [todos, setTodos] = useState(getLocalStorage());
+  const [addTodo, setAddTodo] = useState("");
+  const [editTodo, setEditTodo] = useState("");
+  const [click, setClick] = useState(false);
+
+  useEffect(() => {
+    localStorage.setItem("todos", JSON.stringify(todos));
+  }, [todos]);
+
+  const handleAdd = () => {
+    const todo = {
+      id: uuidv4(),
+      title: addTodo,
+      status: false,
+      date: Date(),
+      isDone: false,
+    };
+    setTodos([todo, ...todos]);
+    setAddTodo("");
+  };
+
+  const handleEdit = () => {
+    const newTodos = todos.filter((t) => t.id !== editTodo.id);
+    setTodos([editTodo, ...newTodos]);
+    setEditTodo("");
+  };
+
+  const handleDelete = (id) => {
+    const todo = todos.filter((t) => t.id !== id);
+    setTodos(todo);
+  };
+
+  const handleDone = (id) => {
+    const todo = todos.map((t) => {
+      if (t.id === id && t.status === true)
+        return { ...t, status: !t.status, isDone: !t.isDone };
+
+      if (t.id === id) return { ...t, isDone: !t.isDone };
+
+      return t;
+    });
+    setTodos(todo);
+  };
```

Comment:
Finns ett snyggt hack du kan göra här istället, om du tar in en hel todo istället för bara id
```suggestion
  const handleDone = (todo) => {
    todo.isDone = !todo.isDone;
    setTodos([...todos]);
  };
```

Reply:
Funkar inte